### ISSUE TITLE
Refactor Authentication in IRS

### ISSUE THREAD

**Issue by jzbmw** [eclipse-tractusx/item-relationship-service#259]
**As** PO,
**I want** to change the authentication of the IRS,
**so that** it is more efficient to integrate the service.

## Hints / Details
- child of eclipse-tractusx/sig-release#218
- Add information to connect to own idp service within the deployment context
- Change from client_id client_secret to api key authentication 
- think about the two users within the IRS (user, admin)
- https://miro.com/app/board/uXjVOGBahIA=/?moveToWidget=3458764569152538759&cot=14
- OAuth2 option is explained where to change code of the application. Keep as much usable Code for OAuth2 as possible. 
- If anyone wants to use OAuth2 then they will have to adapt that in the code base for their deployment. It is not being shipped in an IRS chart

### Outcome
- Documentation has been changed -> https://catenax-ng.github.io/tx-item-relationship-service/docs/arc42/full.html#_safety_and_security_concepts
 - hint was added what is necessary to do if IRS runs in a standalone company context!
- Check and refactor documentation if roles and rights reflect the current situation

### Acceptance Criteria
- [x] Auth process has been changed to API_Keys (Different API Keys for different roles)
- [x] IRS has APIKeys enabled per default  
- [x] IRS OAuth2 Authentication is not able to be used via configuration
- [x] IRS OAuth2  linked to a tag where OAuth2 is integrated into the documentation. 
- [x] Documentation is updated accordingly 
- [x] two roles, two token

## Out of Scope

- ...

**Comment by jzbmw** [comment]
Blocked by Meeting, of which we need input on how to change this.

**Comment by jzbmw** [comment]
As described in the Acceptance Criteria

**Comment by ds-alexander-bulgakov** [comment]
Update after several tryouts and retests: Change to API-Keys is working fine on IRS, all requests can be sent out and either tavern- and cucumber-tests can be run as expected.
Currently ESS-INT has authentication issues, this is investigated by @ds-ext-kmassalski. Test ongoing.

**Comment by ds-alexander-bulgakov** [comment]
**Update**: After the INT_ADMIN_USER secrets were updated with correct values in [#397](https://github.com/eclipse-tractusx/sig-infra/issues/397) all tests could be run as without authentication errors.
Documentation has been updated.
